File: sumOfDivisors.py
def sum(n):
    if n==1:
        return 1
    s=0

    for i in range(1,int(n**0.5)+1):
        if n%i==0:
            s+=i
            if i!=n//i:
                s+=n//i
    return s

n=10
print(sum(n))

File: Maths/sumOfDivisors.java
package Maths;

public class sumOfDivisors {
    public static void main(String[] args) {
        int n = 12;
        System.out.println(sumDivisors(n));
    }

    public static int sumDivisors(int n) {
        // int sum = 0;
        // for (int i = 1; i < n; i++) {
        //     if (n % i == 0) {
        //         sum += i;
        //     }
        // }
        int sum = 1;
        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0) {
                sum += i;
                if (i * i != n) {
                    sum += n / i;
                }
            }
        }
        return sum;
    }
}
